lib/daemon/index.js

```diff
@@ -1,45 +1,56 @@
 "use strict";
 
-const tap = require("lodash/fp").tap;
 const cpus = require("os").cpus;
 const path = require("path");
 const workerpool = require("workerpool");
 
 const actions = require("../actions");
 const Cache = require("../utils/cache");
-const hash = require("../utils/hash");
 
 class InspectpackDaemon {
   /**
    * Start the daemon with an asynchronously initialized cache.
    *
    * @param {Object} opts                  Object options
-   * @param {Object} opts.cacheFilename    The filename of the daemon cache
+   * @param {Object} opts.cache            A cache instance
    * @returns {InspectpackDaemon} The daemon with cache
    */
   static create(opts) {
-    opts = opts || {};
-    const cache = Cache.create({ filename: opts.cacheFilename });
-    return new InspectpackDaemon(cache, opts);
+    return new InspectpackDaemon(opts);
   }
 
   /**
    * Start the daemon with either no cache or an existing cache.
    *
-   * @param {Cache} cache                  An existing cache instance
-   * @param {Object} opts                  Object options
-   * @param {Object} opts.cacheFilename    The filename of the daemon cache
+   * @param {Object} opts                 Object options
+   * @param {Object} opts.cache           A cache instance
+   * @param {Object} opts.cacheFilename   A filename to create a cache if not provided.
+   * @returns {InspectpackDaemon}         The daemon with cache
    */
-  constructor(cache, opts) {
-    this._cache = cache || null;
-    this._cacheFilename = opts.cacheFilename || null;
+  constructor(opts) {
+    opts = opts || {};
+    this._cache = opts.cache ||
+      (opts.cacheFilename ? Cache.create({ filename: opts.cacheFilename }) : null);
+    if (!this._cache) {
```

Comment:
This works because of the noop cache, right?

Reply:
Yep!